src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,587 @@
+//! if you wanted to find the interval histogram for 4000 seconds into the benchmark, you would load
+//! the second hour's file, add each interval's timestamp to that log's BaseTime, and select the one
+//! whose (timestmap + BaseTime) was 4000 bigger than the StartTime. This seems to be how the Java
+//! impl uses it: `HistogramLogReader` lets you filter by "non-absolute" start/end time or by
+//! "absolute" start/end time. The former uses a range of deltas from StartTime and selects
+//! intervals where `interval_timestamp + base_time - start_time` is in the requested range, while
+//! the latter uses a range of absolute timestamps and selects via `interval_timestamp + base_time`.
+//!
+//! ### Timestamp recommendations
+//!
+//! As you can see from that slab of text, using both BaseTime and StartTime is complex.
+//!
+//! We suggest one of the following:
+//!
+//! - Don't use a timestamp header, and simply have each interval's timestamp be the seconds since
+//! epoch.
+//! - Use StartTime, and have each interval's timestamp be a delta from StartTime.
+//!
+//! Of course, if you are writing logs that need to work with an existing log processing pipeline,
+//! you should use timestamps as expected by that logic, so we provide the ability to have all
+//! combinations of timestamp headers if need be.
+//!
+//! # Examples
+//!
+//! Parse a single interval from a log.
+//!
+//! ```
+//! use hdrsample::serialization::interval_log;
+//!
+//! // two newline-separated log lines: a comment, then an interval
+//! let log = b"#I'm a comment\nTag=t,0.127,1.007,2.769,base64EncodedHisto\n";
+//!
+//! let mut iter = interval_log::IntervalLogIterator::new(&log[..]);
+//!
+//! // the comment is consumed and ignored by the parser, so the first event is an Interval
+//! match iter.next().unwrap().unwrap() {
+//!     interval_log::LogEntry::Interval(h) => {
+//!         assert_eq!(0.127, h.start_timestamp());
+//!     }
+//!     _ => panic!()
+//! }
+//!
+//! // there are no more lines in the log; iteration complete
+//! assert_eq!(None, iter.next());
+//! ```
+//!
+//! Skip logs that started before 3 seconds.
+//!
+//! ```
+//! use hdrsample::serialization::interval_log;
+//!
+//! let mut log = Vec::new();
+//! log.extend_from_slice(b"#I'm a comment\n");
+//! log.extend_from_slice(b"Tag=a,0.123,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"1.456,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"3.789,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"Tag=b,4.123,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"5.456,1.007,2.769,base64EncodedHisto\n");
+//! log.extend_from_slice(b"#Another comment\n");
+//!
+//! let iter = interval_log::IntervalLogIterator::new(&log);
+//!
+//! let count = iter.map(|r| r.unwrap())
+//!     // only look at intervals (which are the only non-comment lines in this log)
+//!     .filter_map(|e| match e {
+//!         interval_log::LogEntry::Interval(ilh) => Some(ilh),
```

Comment:
I would match on `Ok(interval_log::LogEntry::Interval(ilh))` here to avoid the `unwrap` just above.

Reply:
Without having read further down, I also don't see why the `IntervalLogIterator` yields a `Result`. What is the `Err` case? And wouldn't it be easier to use with the errors being another `LogEntry` type? Or does the iterator behave weirdly once an `Err` has been yielded?

Reply:
I agree with all the above.